modules/core/shared/src/test/scala/gem/laws/SubgroupLaws.scala

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.laws
+
+// A is a subgroup of B if covariant widening is a group homomorphism.
+abstract class SubgroupLaws[A, B](f: A <:< B) extends GroupHomomorphismLaws[A, B](f)
```

Comment:
I assume that you mean by "covariant widening" that you're just talking about the natural map that embeds `A` into `B`. I'm not sure exactly what `<:<` means.

Reply:
I'm assuming in this case that you are considering that `A` is a subgroup (not necessarily proper) of `B`?

You're not considering, say, the group `A` of all the multiples of, say, the 2x2 identity matrix under addition  a direct subgroup of the rationals even though you can map `A` isomorphically to the integers, which clearly have an embedding homomorphism into the rationals?

Reply:
Yes. In Scala terms

- if `A` is a subtype of `B` (a value of type `A <:< B` is a proof of this fact), and
- we have `Group[A]` and `Group[B]`, and
- we claim that `A` is a subgroup of `B`, then

the widening `A => B` (i.e., just forgetting it's an `A`) must be a group homomorphism.

Reply:
> You're not considering ...

Yes, if I understand what you're saying. Here every `A` *is* a `B`; it's a subset. So this isn't an arbitrary mapping.

Reply:
That sounds exactly right to me. Thanks for the explanation regarding `<:<`.